TESTPLAN.md

```diff
@@ -1,161 +1,112 @@
-
-- prove the notification bar ui opens
-- *clicking on the left-most button presented*.
-- verifying that sent Telemetry is correct.
-
-Code at [./test/functional_test.js].
 
 ## Manual / QA TEST Instructions
 
-Assumptions / Thoughts
-
-1.  Please ask if you want  more command-line tools to do this testing.
-
-### BEFORE EACH TEST: INSTALL THE ADDON to a CLEAN (NEW) PROFILE
-
-0.  (create profile:  https://developer.mozilla.org/en-US/Firefox/Multiple_profiles, or via some other method)
-1.  In your Firefox profile
-2.  `about:debugging` > `install temporary addon`
-
-As an alternative (command line) cli method:
-
-1. `git clone` the directory.
-2. `npm install` then `npm run firefox` from the Github (source) directory.
-
-
-### Note: checking "Correct Pings"
-
-All interactions with the UI create sequences of Telemetry Pings.
-
-All UI `shield-study` `study_state` sequences look like this:
-
-- `enter => install => (one of: "voted" | "notification-x" |  "window-or-fx-closed") => exit`.
-
-(Note: this is complicated to explain, so please ask questions and I will try to write it up better!, see `TELMETRY.md` and EXAMPLE SEQUENCE below.)
-
-### Do these tests.
-
-1.  UI APPEARANCE.  OBSERVE a notification bar with these traits:
-
-    *  Icon is 'heartbeat'
-    *  Text is one of 8 selected "questions", such as:  "Do you like Firefox?".  These are listed in [./addon/Config.jsm] as the variable `weightedVariations`.
-    *  clickable buttons with labels 'yes | not sure | no'  OR 'no | not sure | yes' (50/50 chance of each)
-    *  an `x` button at the right that closes the notice.
-
-    Test fails IF:
+### PREREQUISITES:
 
-    - there is no bar.
-    - elements are not correct or are not displayed
+    Node 8+ w/ npm 5
+    Firefox Nightly
 
+### Installation:
 
-2.  UI functionality: VOTE
+    git clone https://github.com/gregglind/addon-wr.git
+    cd addon-wr
+    npm install
 
-    Expect:  Click on a 'vote' button (any of: `yes | not sure | no`) has all these effects
 
-    - notice closes
-    - addon uninstalls
-    - no additional tabs open
-    - telemetry pings are 'correct' with this SPECIFIC `study_state` as the ending
+### Tests and Beliefs:
 
-        - ending is `voted`
-        - 'vote' is correct.
+1.  BUILDABLE.  It always builds.
 
-3.  UI functionality: 'X' button
+   `npm run build` makes an addon:
 
-    Click on the 'x' button.
+    ```
+    -rw-r--r--   1 glind  staff     13 Nov  8 15:36 .gitignore
+    lrwxr-xr-x   1 glind  staff     43 Dec  8 13:09 linked-addon.xpi@ -> pug.experience@shield.mozilla.org-1.0.0.xpi
+    -rw-r--r--   1 glind  staff  21975 Dec  8 13:09 pug.experience@shield.mozilla.org-1.0.0.xpi
+    ```
 
-    - notice closes
-    - addon uninstalls
-    - no additional tabs open
-    - telemetry pings are 'correct' with this SPECIFIC ending
+2.  LINTABLE:  It lints, with **warnings** but **no errors**.
 
-      - ending is `notification-x`
+    `npm run eslint`
```

Comment:
I think we still need to fix the parent `npm run lint` uber-test, since that failed when I checked earlier.

Reply:
npm run eslint   NOT npm run lint

Reply:
ripped them all out.  that was a partial conversion.